### PR TITLE
aws_transit_gateway_connect

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource. No integration tests with terraform are done as there is no terraform resource available.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_transit_gateway_connect.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+Ensure that transit gateway connect exists.
+
+    describe aws_transit_gateway_connect(transit_gateway_attachment_ids: 'tgw-attach-1234567890') do
+      it { should exits }
+    end
+
+## Parameters
+
+`transit_gateway_attachment_ids` _(required)_
```
Comment (Nirbhay1997) [inline docs/resources/aws_transit_gateway_connect.md:22]:
```suggestion
`transit_gateway_attachment_id` _(required)_
```

File: docs/resources/aws_transit_gateway_connect.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [Universal Matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+The controls will pass if the `describe` method returns at least one result.
+
+### exist
+
+Use `should` to test that the entity exists.
+
+    describe aws_transit_gateway_connect(transit_gateway_attachment_ids: 'tgw-attach-1234567890') do
```
Comment (Nirbhay1997) [inline docs/resources/aws_transit_gateway_connect.md:60]:
```suggestion
    describe aws_transit_gateway_connect(transit_gateway_attachment_id: 'tgw-attach-1234567890') do
```
There are multiple places we need to correct this

File: libraries/aws_transit_gateway_connect.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      it { should eq 'test' }
+    end
+
+    describe aws_transit_gateway_connect(transit_gateway_attachment_ids: 'test') do
+      it { should exits }
+    end
+  "
+
+  def initialize(opts = {})
+    opts = { transit_gateway_attachment_ids: opts } if opts.is_a?(String)
```
Comment (Nirbhay1997) [inline libraries/aws_transit_gateway_connect.rb:20]:
```suggestion
    opts = { transit_gateway_attachment_id: opts } if opts.is_a?(String)
```

Comment (Nirbhay1997) [reply libraries/aws_transit_gateway_connect.rb:20]: other places also

**Review by Nirbhay1997 — APPROVED** [review]
I believe there are no tf for this resource 
LGTM

File: docs/resources/aws_transit_gateway_connect.md (hunk 1 above)
Comment (soumyo13) [reply docs/resources/aws_transit_gateway_connect.md:22]: Done.

File: docs/resources/aws_transit_gateway_connect.md (hunk 2 above)
Comment (soumyo13) [reply docs/resources/aws_transit_gateway_connect.md:60]: Done.

File: libraries/aws_transit_gateway_connect.rb (hunk 3 above)
Comment (soumyo13) [reply libraries/aws_transit_gateway_connect.rb:20]: Done.

**Comment by soumyo13** [comment]
Please check. I have pushed the changes as requested.

**Pull request merged**